tests/Nightly/test_llm_runtime.py

```diff
@@ -0,0 +1,79 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import shutil
+import torch
+import unittest
+
+from transformers import AutoTokenizer
+from intel_extension_for_transformers.transformers import AutoModel, WeightOnlyQuantConfig
+
+def cmpData(numa, numb):
+    totalErr = ((np.abs(numa - numb))**2).sum()
+    totalNum = (np.abs(numa)**2).sum()
+    diff2 = np.sqrt(totalErr/totalNum)
+
+    cos = np.dot(numa, numb)/(np.linalg.norm(numa)*np.linalg.norm(numb))
+    return {"diff2": diff2, "cos": cos}
+
+class TestLLMRUNTIME(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        pass
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        shutil.rmtree("./runtime_outs", ignore_errors=True)
+
+    def test_llm_runtime(self):
+        model_name = "/tf_dataset2/models/nlp_toolkit/llama-2-7b-chat/Llama-2-7b-chat-hf"
+        prompt = "What is the meaning of life?"
+
+        tokenizer = AutoTokenizer.from_pretrained(model_name, trust_remote_code=True)
+        inputs = tokenizer(prompt, return_tensors="pt")
+        
+        pt_logits = torch.load("/tf_dataset2/inc-ut/nlptoolkit_ut_model/llama2_pt_logits.pth")[:,-1]
+        pt_generate_ids = torch.load("/tf_dataset2/inc-ut/nlptoolkit_ut_model/llama2_pt_generate_ids.pth")[0].tolist()
+        print(tokenizer.decode(pt_generate_ids))
+
+        # check output ids
+        woq_config = WeightOnlyQuantConfig(use_quant=False)
+        itrex_model = AutoModel.from_pretrained(model_name, quantization_config=woq_config, use_llm_runtime=True, trust_remote_code=True)
+        itrex_generate_ids = itrex_model.generate(inputs.input_ids, do_sample=False, max_new_tokens=100)[0]
+        print(tokenizer.decode(itrex_generate_ids))
+        for i in range(len(pt_generate_ids)):
+            self.assertEqual(pt_generate_ids[i], itrex_generate_ids[i])
+
+        # check diff of logits
+        itrex_model = AutoModel.from_pretrained(model_name, load_in_4bit=True, 
+                                                use_llm_runtime=True, trust_remote_code=True)
+        itrex_logits = itrex_model(inputs.input_ids)
+        cmp = cmpData(pt_logits.detach().numpy().flatten(), itrex_logits.flatten())
+        print("load_in_4bit: ", cmp)
+        self.assertTrue(cmp["diff2"] < 0.5)
+
+        itrex_model = AutoModel.from_pretrained(model_name, load_in_8bit=True, 
+                                                use_llm_runtime=True, trust_remote_code=True)
+        itrex_logits = itrex_model(inputs.input_ids)
+        cmp = cmpData(pt_logits.detach().numpy().flatten(), itrex_logits.flatten())
+        print("load_in_8bit: ", cmp)
+        self.assertTrue(cmp["diff2"] < 0.5)
```

Comment:
0.5 seems a weak checker. @zhenwei-intel  should be more strict e.g., 1e-5?

Reply:
The diff value of fp32 is indeed below 1e-5. I compared the output token id of fp32 directly with pytorch.

For 4bit and 8bit, the diff value will be larger, and basically > 0.1.

There is not yet enough data to set this threshold. However, the diff value exceeding 0.5 will definitely be wrong outputs.

Reply:
@hshen14 updated the threshold, thanks for suggestion

Reply:
Llama-2-7b-chat-hf/
fp32 {'diff2': 0.00034225243, 'cos': 1.0}
ggml_int4 {'diff2': 0.16917692, 'cos': 0.98622376}
jblas_int4 {'diff2': 0.2865865, 'cos': 0.9583879}
jblas_int8 {'diff2': 0.24928005, 'cos': 0.9685063}